### PR TITLE
Add XOAUTH2 authentication

### PR DESCRIPTION
**Pull request by robstewart57**
Adds OAuth based authentication.

Fixes https://github.com/awdeorio/mailmerge/issues/150

### CODE REVIEW

File: mailmerge/sendmail_client.py [hunk 1]
```diff
@@ -104,6 +104,19 @@ def sendmail(self, sender, recipients, message):
...
                     smtp.sendmail(sender, recipients, message_flattened)
+            elif self.config.security == "XOAUTH":
+                with smtplib.SMTP(host, port) as smtp:
+                    smtp.ehlo()
+                    smtp.starttls()
+                    smtp.ehlo()
+                    access_token = self.password
+                    C_A = b'\x01'
+                    user = ("user=" + self.config.username).encode("ascii")
+                    btoken = ("auth=Bearer " + access_token).encode("ascii")
```
[the remark this answers is not in the document]
Comment (robstewart57) [reply mailmerge/sendmail_client.py:115]: @awdeorio Exception handling added: https://github.com/awdeorio/mailmerge/pull/152/commits/7f31846cee349f8236d269785bb732df9f2dd7a9

**Comment by robstewart57** [comment]
> Is ASCII required?

Yes, without `.encode("ascii")` this type error is thrown:

> TypeError: can only concatenate str (not "bytes") to str

I've added exception handling here: https://github.com/awdeorio/mailmerge/commit/7f31846cee349f8236d269785bb732df9f2dd7a9

> Would you be willing to add a test? You could copy paste [test_security_starttls](https://github.com/awdeorio/mailmerge/blob/develop/tests/test_sendmail_client.py#L211) as a starting point.

I'm unfamiliar with Python tooling, so I'm unsure how to run the existing tests.

1. How do I install the dependencies for running the tests?
2. How do I run the existing tests?
3. To add a test for this XOAUTH authentication would it literally be copy/pasting the `test_security_starttls` test?
4. Do I need to add this new test to a list of tests anywhere?

**Comment by awdeorio** [comment]
Thanks for adding the exception handling.  To install deps and run the unit tests check out [these commands](https://github.com/awdeorio/mailmerge/blob/develop/CONTRIBUTING.md#install-development-environment).  You don't need to add anything to a list, just copy paste an old test, change the name, and modify the code a bit.

**Comment by robstewart57** [comment]
Hi @awdeorio , the `test_security_xoauth` test is added in https://github.com/awdeorio/mailmerge/pull/152/commits/406716c58485fd8cc65a174fc52c1744e33c8aae .

**Comment by robstewart57** [comment]
> Would you be able to test this updated code on a real server just to be extra sure?

@awdeorio I've tested this branch with your latest commits, against smtp.office365.com on port 587.

It sends emails as expected, so I think we're good to go.

**Comment by awdeorio** [comment]
Great!  Merging!

**Pull request merged**